### DISCUSSION TITLE
Lazy Loading option

### DISCUSSION THREAD

**Discussion by midweste** [stayallive/wp-sentry#118] [Ideas]
It seems that sentry is addressing some performance issues with fully loading Sentry and is allowing for lazy-loading of the js component until after page load (or when initiated). https://github.com/getsentry/sentry-javascript/issues/2233#issuecomment-547377460

https://snipboard.io/lufAWJ.jpg

This post is marked version 6.16.1 so maybe a future update as it looks like the js component is @sentry/browser 6.15.0 (aa30ab6)

https://docs.sentry.io/platforms/javascript/install/lazy-load-sentry/

If I have some time over the holiday, will take a look and try to provide a pull request.  Thank you for all your efforts!

**Comment by stayallive** [comment]
I'm not sure we can do much, it looks like this requires the use of the sentry.io CDN.

Once we go this far into it it might be better to "just" include that script yourself instead of using the JS stuff in this plugin since it doesn't add too much.

But happy to take a look at suggestions on how this would have to look in the future 👍

**Comment by midweste** [comment]
Will do a little experimenting.  Think I found the loader script in the source for sentry.  Will shoot for adding an option to the plugins page for lazy loading and include this loader script instead of the full sentry script

https://github.com/getsentry/sentry-javascript/blob/master/packages/browser/src/loader.js

**Reply by stayallive** [reply]
Sounds good, except we don't have an options page and are also not planning for one so it will have to be a option you set in the `wp-config.php` file like the rest. 

But let me know what you come up with. Happy to take a look if it's a good fit for this plugin 🙏

**Reply by midweste** [reply]
We can continue discussion here https://github.com/stayallive/wp-sentry/pull/119
